db/migrate/20240501171850_add_index_to_dependencies.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class AddIndexToDependencies < ActiveRecord::Migration[7.0]
+  disable_ddl_transaction!
+
+  def change
+    # this has already been run on production
+    return if Rails.env.production?
+
+    # the covering index (project_id, ((created_at)::date)) already enables this
+    remove_index :dependencies, :project_id, if_exists: true, algorithm: :concurrently
+    add_index :dependencies, %i[project_id version_id], algorithm: :concurrently
```

Comment:
I'm not 100% sure this will help all cases, but it did seem to help with a fairly popular project:

```
p = Project.find_by platform: "Rubygems", name: "devise"

# BEFORE
> Benchmark.ms { Version.where(id: p.dependents.select(:version_id)).select(:project_id).distinct.count }
=> 4299.041126971133

# AFTER
> Benchmark.ms { Version.where(id: p.dependents.select(:version_id)).select(:project_id).distinct.count }
=> 127.43865209631622
```

The query plans don't seem too different cost-wise, but the new one does seem simpler:

# BEFORE

<img width="781" alt="Screenshot 2024-05-01 at 12 27 05 PM" src="https://github.com/librariesio/libraries.io/assets/5054/af9cdb2f-100a-491f-8e9e-21cedd49a5be">


# AFTER


<img width="865" alt="Screenshot 2024-05-01 at 12 26 59 PM" src="https://github.com/librariesio/libraries.io/assets/5054/48f34e36-9aca-4b45-884c-fc5d1c24350f">